rs/cross-chain/proposal-cli/src/main.rs

```diff
@@ -153,45 +184,66 @@ fn write_to_disk<P: Into<ProposalTemplate>>(
     }
     fs::create_dir_all(&output_dir).unwrap_or_else(|_| panic!("failed to create {:?}", output_dir));
 
-    let args_file_path = output_dir.join("args.bin");
-    let mut args_file = fs::File::create(&args_file_path)
-        .unwrap_or_else(|_| panic!("failed to create {:?}", args_file_path));
+    let bin_args_file_path = output_dir.join("args.bin");
+    let mut args_file = fs::File::create(&bin_args_file_path)
+        .unwrap_or_else(|_| panic!("failed to create {:?}", bin_args_file_path));
     proposal.write_bin_args(&mut args_file);
     println!(
         "Binary upgrade args written to '{}'",
-        args_file_path.display()
+        bin_args_file_path.display()
     );
 
-    let args_file_path = output_dir.join("args.hex");
-    let mut args_file = fs::File::create(&args_file_path)
-        .unwrap_or_else(|_| panic!("failed to create {:?}", args_file_path));
+    let hex_args_file_path = output_dir.join("args.hex");
+    let mut args_file = fs::File::create(&hex_args_file_path)
+        .unwrap_or_else(|_| panic!("failed to create {:?}", hex_args_file_path));
     proposal.write_hex_args(&mut args_file);
     println!(
         "Hexadecimal upgrade args written to '{}'",
-        args_file_path.display()
+        hex_args_file_path.display()
     );
 
     let artifact = output_dir.join(proposal.target_canister().artifact_file_name());
     ic_repo.copy_file(&proposal.target_canister().artifact(), &artifact);
     println!("Artifact written to '{}'", artifact.display());
 
-    let proposal = proposal.render();
-    if proposal.len() > GOVERNANCE_PROPOSAL_SUMMARY_BYTES_MAX {
+    let proposal_summary_content = proposal.render();
+    if proposal_summary_content.len() > GOVERNANCE_PROPOSAL_SUMMARY_BYTES_MAX {
         errors.push(format!(
             "Proposal summary is too long and will fail validation from the governance canister when submitted: {} bytes (max {})",
-            proposal.len(),
+            proposal_summary_content.len(),
             GOVERNANCE_PROPOSAL_SUMMARY_BYTES_MAX
         ));
     }
     let proposal_summary = output_dir.join("summary.md");
     let mut summary_file = fs::File::create(&proposal_summary)
         .unwrap_or_else(|_| panic!("failed to create {:?}", proposal_summary));
-    summary_file.write_all(proposal.as_bytes()).unwrap();
+    summary_file
+        .write_all(proposal_summary_content.as_bytes())
+        .unwrap();
```

Comment:
Are these `unwrap` calls safe? Or should the error cases be handled (properly)?

Reply:
> Are these unwrap calls safe?

I'm not sure what you mean with "safe". The code could panic, if for example the process doesn't have write permissions in the given directory (whose path is specified by the user). However, generally panicking in this tool is IMO acceptable because it's an internal tool to generate boilerplate files to make proposals, which will be reviewed by the community when submitted. 

> Or should the error cases be handled (properly)?

For that concrete case, there is not much we can do, if we cannot create the file. We could just skip it if there was an error instead of panicking, but I think here panicking is better because the caller is at least aware that there was a problem. Does-it make sense?

Reply:
Resolving thread to merge. Feel free to ping me if there is any unclarity left.